### PR TITLE
Added second paragraph to Stack traces option

### LINKED ISSUE #1235: [DOC] Update error_trace URL parameter description
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
Coming from this issue[#4255](https://github.com/opensearch-project/OpenSearch/issues/4255)
The description of Stack traces in [OpenSearch documentation](https://opensearch.org/docs/latest/opensearch/common-parameters/) is not comprehensive and may leading to some misunderstanding. 

"To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL." is right.
It is better to add the description of Config setting: If there is a setting of `http.detailed_errors.enabled: false` in config file, the `error_trace` url request will be rejected. 

The detailed logic will be: 
When the config setting `http.detailed_errors.enabled: false`, the request of `error_trace=true` will not contain the detailed error message.
```
$ curl "http://localhost:9200/opensearch/_search?error_trace=false"
{"error":"IndexNotFoundException[no such index [opensearch]]","status":404}

$ curl "http://localhost:9200/opensearch/_search?error_trace=true" 
{"error":"error traces in responses are disabled.","status":400}   
```
This code regarding to this setting is from: https://github.com/opensearch-project/OpenSearch/blob/2.2.1/server/src/main/java/org/opensearch/rest/RestController.java#L368-L375

When there is no config setting for `http.detailed_errors.enabled` or `http.detailed_errors.enabled: true`, the response of URL parameter will contain detailed error message:
```
$ curl "http://localhost:9200/opensearch/_search?error_trace=true&pretty"
{
  "error" : {
    "root_cause" : [
      {
        "type" : "index_not_found_exception",
        "reason" : "no such index [opensearch]",
        "index" : "opensearch",
        "resource.id" : "opensearch",
        "resource.type" : "index_or_alias",
        "index_uuid" : "_na_",
        "stack_trace" : "[opensearch] IndexNotFoundException[no such index [opensearch]]\n\tat org.opensearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.indexNotFoundException(IndexNameExpressionResolver.java:1067)\n\tat org.opensearch.cluster.metadata.IndexNameExpressionResolver$WildcardExpressionResolver.innerResolve(IndexNameExpressionResolver.java:1004)
```

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
Added text: "When the config setting http.detailed_errors.enabled: false, the request of error_trace=true will not contain the detailed error message."

### Issues Resolved
Fixes #1235 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/common-parameters.md [hunk 1]
```diff
@@ -14,4 +14,4 @@ Human-readable output | To convert output units to human-readable values (for ex
...
 Content type | To specify the type of content in the request body, use the `Content-Type` key name in the request header. Most operations support JSON, YAML, and CBOR formats. | `POST _scripts/<template_name> -H 'Content-Type: application/json`
 Request body in query string | If the client library does not accept a request body for non-POST requests, use the `source` query string parameter to pass the request body. Also, specify the `source_content_type` parameter with a supported media type such as `application/json`. | `GET _search?source_content_type=application/json&source={"query":{"match_all":{}}}`
-Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. | `GET <index_name>/_search?error_trace=true`
+Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. <br> <br> When the config setting `ttp.detailed_errors.enabled: false`, the request of `error_trace=true` will not contain the detailed error message.| `GET <index_name>/_search?error_trace=true`
```
Comment (Naarcha-AWS) [inline _opensearch/common-parameters.md:17]:
```suggestion
Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. <br> <br> When the configuration setting `ttp.detailed_errors.enabled:` is set to `false` in opensearch.yml, `error_trace=true` will not contain a detailed error message.| `GET <index_name>/_search?error_trace=true`
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _opensearch/common-parameters.md (hunk 1 above)
Comment (kolchfa-aws) [reply _opensearch/common-parameters.md:17]: Is there a reason why "stack traces" is plural here? Normally, especially if the only option is error stack trace, it is singular.

Comment (kolchfa-aws) [reply _opensearch/common-parameters.md:17]: Also, please remove the colon in the suggestion: `ttp.detailed_errors.enabled:` should be `ttp.detailed_errors.enabled`

Comment (Naarcha-AWS) [reply _opensearch/common-parameters.md:17]: @kolchfa-aws: To look at this from all angles, is it for when you're using this parameter in the context of a response that can return multiple indexes and thus multiple stack traces, such as `https://opensearch.org/docs/latest/opensearch/rest-api/document-apis/multi-get/`?

Still for changing it though.

Comment (kolchfa-aws) [reply _opensearch/common-parameters.md:17]: Ah ok. That makes sense. In that case, we can go either way. I would still change to singular though.

**Review by Naarcha-AWS — APPROVED** [review]

**Review by kolchfa-aws — CHANGES_REQUESTED** [review]
Please change to: When the configuration setting `http.detailed_errors.enabled`. Add the 'h' and remove the ':'

File: _opensearch/common-parameters.md [hunk 2]
```diff
@@ -14,4 +14,4 @@ Human-readable output | To convert output units to human-readable values (for ex
...
 Content type | To specify the type of content in the request body, use the `Content-Type` key name in the request header. Most operations support JSON, YAML, and CBOR formats. | `POST _scripts/<template_name> -H 'Content-Type: application/json`
 Request body in query string | If the client library does not accept a request body for non-POST requests, use the `source` query string parameter to pass the request body. Also, specify the `source_content_type` parameter with a supported media type such as `application/json`. | `GET _search?source_content_type=application/json&source={"query":{"match_all":{}}}`
-Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. | `GET <index_name>/_search?error_trace=true`
+Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. <br> <br> When the configuration setting `ttp.detailed_errors.enabled:` is set to `false` in opensearch.yml, `error_trace=true` will not contain a detailed error message.| `GET <index_name>/_search?error_trace=true`
```
Comment (Naarcha-AWS) [inline _opensearch/common-parameters.md:17]:
```suggestion
Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. <br> <br> When the configuration setting `http.detailed_errors.enabled:` is set to `false` in opensearch.yml, `error_trace=true` will not contain a detailed error message.| `GET <index_name>/_search?error_trace=true`
```

**Review by Naarcha-AWS — COMMENTED** [review]
Per Fanit's comment.

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _opensearch/common-parameters.md [hunk 3]
```diff
@@ -14,4 +14,4 @@ Human-readable output | To convert output units to human-readable values (for ex
...
 Content type | To specify the type of content in the request body, use the `Content-Type` key name in the request header. Most operations support JSON, YAML, and CBOR formats. | `POST _scripts/<template_name> -H 'Content-Type: application/json`
 Request body in query string | If the client library does not accept a request body for non-POST requests, use the `source` query string parameter to pass the request body. Also, specify the `source_content_type` parameter with a supported media type such as `application/json`. | `GET _search?source_content_type=application/json&source={"query":{"match_all":{}}}`
-Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. | `GET <index_name>/_search?error_trace=true`
+Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. <br> <br> When the configuration setting `http.detailed_errors.enabled` is set to `false` in opensearch.yml, `error_trace=true` will not contain a detailed error message.| `GET <index_name>/_search?error_trace=true`
```
Comment (rursprung) [inline _opensearch/common-parameters.md:17]: note that in that case it'll actually fail with an error telling you that you're not allowed to set `error_trace=true`

**Pull request merged**